### PR TITLE
Added upgradability test in deploy scripts

### PR DESCRIPTION
**Pull request by YosephKS**
- Added back legacy V1 contracts for MoxieBondingCurve.sol, Staking.sol, and SubjectFactory.sol to `contracts/test` folder
- Add the deployment of legacy V1 contracts in the `deploy` function to simulate contract upgrading for each test cases

### CODE REVIEW

File: packages/protocol/test/MoxieBondingCurve.ts [hunk 1]
```diff
@@ -1,4745 +1,4753 @@
 import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
 import { expect } from "chai";
-import hre, { ethers } from "hardhat";
-import { SubjectERC20 as SubjectERC20Type } from "../typechain-types";
+import hre, { ethers, upgrades } from "hardhat";
+import type { SubjectERC20 as SubjectERC20Type } from "../typechain-types";
 import {
-    getExpectedSellReturnAndFee,
-    getExpectedBuyAmountAndFee,
+  getExpectedSellReturnAndFee,
```
Comment (0xsarvesh) [inline packages/protocol/test/MoxieBondingCurve.ts:6]: I think you can create a new test file `MoxieBondingCurveUpgradeTest.ts` which covers upgradability test

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]
Please rebase your branch , as old contracts are added.

File: packages/protocol/test/MoxieBondingCurve.ts (hunk 1 above)
Comment (YosephKS) [reply packages/protocol/test/MoxieBondingCurve.ts]: yeah rebased done, I've moved all the upgradability tests to newly created files

File: packages/protocol/contracts/interfaces/IMoxieBondingCurveLegacyV1.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+interface IMoxieBondingCurveLegacyV1 {
+    struct FeeInput {
+        uint256 protocolBuyFeePct;
+        uint256 protocolSellFeePct;
+        uint256 subjectBuyFeePct;
+        uint256 subjectSellFeePct;
+    }
+
+    function initializeSubjectBondingCurve(
```
Comment (0xsarvesh) [inline packages/protocol/contracts/interfaces/IMoxieBondingCurveLegacyV1.sol:12]: Do we need this interface? Its added back in referral branch

Comment (YosephKS) [reply packages/protocol/contracts/interfaces/IMoxieBondingCurveLegacyV1.sol:12]: my bad for got to remove the legacy interfaces, removing them quickly

Comment (YosephKS) [reply packages/protocol/contracts/interfaces/IMoxieBondingCurveLegacyV1.sol]: done @0xsarvesh

**Review by 0xsarvesh — APPROVED** [review]

**Review by sharathkrml — APPROVED** [review]

**Pull request merged**